datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaires.kt

```diff
@@ -114,23 +123,37 @@ private fun validateLaunchContextExtension(launchExtension: Extension) {
   }
 }
 
+/**
+ * Gets the launch contexts resource map, filtered based on the matching nameCode values found in
+ * the launchContextExtensions list.
+ */
+internal fun getMatchingLaunchContexts(
```

Comment:
can we rewrite the function to be a bit more self-explanatory:

```suggestion
/**
 * Filters the provided launch contexts by matching the keys with the `code` values
 * found in the "name" extensions.
 */
internal fun filterLaunchContextsByCodeInNameExtension(
  allLaunchContexts: Map<String, Resource>,
  extensions: List<Extension>
): Map<String, Resource> {
  val codesInNameExtensions = extensions.mapNotNull { extension ->
    (extension.getExtensionByUrl("name").value as? Coding)?.code
  }

  return allLaunchContexts.filterKeys(codesInNameExtensions::contains)
}
```